src/components/transactions/Switch/SwitchModal.tsx

```diff
@@ -1,31 +1,120 @@
 import { Trans } from '@lingui/macro';
 import { Box, CircularProgress, Typography } from '@mui/material';
-import { ContractCallContext, ContractCallResults, Multicall } from 'ethereum-multicall';
-import { providers } from 'ethers';
-import { formatUnits } from 'ethers/lib/utils';
+import { useQueryClient } from '@tanstack/react-query';
 import React, { useEffect, useMemo, useState } from 'react';
 import { ConnectWalletButton } from 'src/components/WalletConnection/ConnectWalletButton';
+import { TokenInfoWithBalance, useTokensBalance } from 'src/hooks/generic/useTokensBalance';
 import { ModalType, useModalContext } from 'src/hooks/useModal';
+import { useWeb3Context } from 'src/libs/hooks/useWeb3Context';
 import { useRootStore } from 'src/store/root';
+import { queryKeysFactory } from 'src/ui-config/queries';
 import { TOKEN_LIST, TokenInfo } from 'src/ui-config/TokenList';
-import {
-  CustomMarket,
-  getNetworkConfig,
-  getProvider,
-  marketsData,
-} from 'src/utils/marketsAndNetworksConfig';
+import { CustomMarket, getNetworkConfig, marketsData } from 'src/utils/marketsAndNetworksConfig';
 
 import { BasicModal } from '../../primitives/BasicModal';
 import { supportedNetworksWithEnabledMarket } from './common';
 import { SwitchModalContent } from './SwitchModalContent';
 
 const defaultNetwork = marketsData[CustomMarket.proto_mainnet_v3];
 
-export interface TokenInfoWithBalance extends TokenInfo {
-  balance: string;
+interface SwitchModalContentWrapperProps {
+  user: string;
+  chainId: number;
+  setSelectedChainId: (chainId: number) => void;
 }
 
+const getFilteredTokens = (chainId: number): TokenInfoWithBalance[] => {
+  let customTokenList = TOKEN_LIST.tokens;
+  const savedCustomTokens = localStorage.getItem('customTokens');
+  if (savedCustomTokens) {
+    customTokenList = customTokenList.concat(JSON.parse(savedCustomTokens));
+  }
+  const transformedTokens = customTokenList.map((token) => {
+    return { ...token, balance: '0' };
+  });
+  const realChainId = getNetworkConfig(chainId).underlyingChainId ?? chainId;
+  return transformedTokens.filter((token) => token.chainId === realChainId);
+};
+
+const SwitchModalContentWrapper = ({
+  user,
+  chainId,
+  setSelectedChainId,
+}: SwitchModalContentWrapperProps) => {
+  const filteredTokens = useMemo(() => getFilteredTokens(chainId), [chainId]);
+
+  const { data: baseTokenList } = useTokensBalance(filteredTokens, chainId, user);
+
+  const { defaultInputToken, defaultOutputToken } = useMemo(() => {
+    if (baseTokenList) {
+      const defaultInputToken =
+        baseTokenList.find((token) => token.extensions?.isNative) || baseTokenList[0];
+      const defaultOutputToken =
+        baseTokenList.find(
+          (token) =>
+            (token.address === AaveV3Ethereum.ASSETS.GHO.UNDERLYING || token.symbol == 'AAVE') &&
+            token.address !== defaultInputToken.address
+        ) || baseTokenList[0];
```

Comment:
I think instead of defaulting to the first item in the token list, should do
```ts
baseTokenList.find(token => token.address !== defaultInputToken.address)
```
so we don't end up with the same asset selected in both inputs

Reply:
i agree, this should be a better solution, i will update